sympy/physics/mechanics/kane.py

```diff
@@ -638,7 +638,7 @@ def forcing_full(self):
         if not self._fr or not self._frstar:
             raise ValueError('Need to compute Fr, Fr* first.')
         f1 = self._k_ku * Matrix(self.u) + self._f_k
-        return -Matrix([f1, self._f_d, self._f_dnh])
+        return -Matrix([f1.subs(self.kindiffdict()), self._f_d.subs(self.kindiffdict()), self._f_dnh.subs(self.kindiffdict())])
```

Comment:
I think `xreplace()` should, in general, be used instead of subs.

Reply:
This also needs a unit test.

Reply:
I'm also not sure that this is right place to do this. It could have quite the affect on speed of computation.

This is the kind of thing that needs to be in a separate PR.

Reply:
I shall use xreplace and add the test. There doesn't seem to be any difference in the runtime of the current mechanics tests though.

Reply:
This gets exacerbated on larger problems.

Reply:
I am not sure about a workaround. I shall move this to another PR then.